sound/soc/sof/intel/hda-dsp.c

```diff
@@ -393,6 +393,10 @@ static int hda_dsp_update_d0i3c_register(struct snd_sof_dev *sdev, u8 value)
 static int hda_dsp_set_D0_state(struct snd_sof_dev *sdev,
 				const struct sof_dsp_power_state *target_state)
 {
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
```

Comment:
I don't think this file can be compiled without SOF_HDA being enabled - with the current Kconfig at least?

Reply:
Ok, that would be the case if `SND_SOC_SOF_HDA_LINK` isn't selected, e.g. for `NOCODEC`. Got it, thanks!